### PR TITLE
Add all the properties for MongoDB Service Binding

### PR DESCRIPTION
**Pull request by sgahlot**
DBAAS-42: Added all the properties for MongoDB. Fixed an issue with trailing slash before options + handling host and port in one property. Setting Quarkus user/pass separately from connection string.
Also changed the type comparison by ignoring case

### CODE REVIEW

File: extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBinding.java [hunk 1]
```diff
@@ -67,14 +68,11 @@ public ServiceBinding(Path bindingDirectory) {
...
 
-        File[] files = directory.toFile().listFiles(new FileFilter() {
-            @Override
-            public boolean accept(File f) {
-                try {
-                    return !Files.isHidden(f.toPath()) && !Files.isDirectory(f.toPath());
-                } catch (IOException e) {
-                    throw new IllegalStateException("Unable to determine if file '" + f + "' is a regular file", e);
-                }
+        File[] files = directory.toFile().listFiles(f -> {
```
Comment (gsmet) [inline extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBinding.java:71]: This is intentional. We avoid lambdas in runtime code as they have a cost. Please revert this change.

File: extensions/kubernetes-service-binding/runtime/src/test/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBindingTest.java [hunk 2]
```diff
@@ -40,4 +43,19 @@ void testK8s() {
         assertThat(binding.getProperties()).containsExactly(entry("test-secret-key", "test-secret-value"));
     }
 
-}
+    @Test
+
```
Comment (gsmet) [inline extensions/kubernetes-service-binding/runtime/src/test/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBindingTest.java:47]: Unnecessary new line.

**Review by gsmet — DISMISSED** [review]
Thanks. I added two comments on the form.

Given these changes are quite invasive, I think a discussion before doing this work would have been welcome. Did you discuss this before with the Quarkus community?

File: extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBinding.java (hunk 1 above)
Comment (sgahlot) [reply extensions/kubernetes-service-binding/runtime/src/main/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBinding.java]: I've reverted this change.

File: extensions/kubernetes-service-binding/runtime/src/test/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBindingTest.java (hunk 2 above)
Comment (sgahlot) [reply extensions/kubernetes-service-binding/runtime/src/test/java/io/quarkus/kubernetes/service/binding/runtime/ServiceBindingTest.java]: Reverted this change as well as the class under test is reverted.

**Comment by sgahlot** [comment]
The only changes in this PR now are related to MongoDB binding. For the k8s binding change, I'll open up a discussion regarding the type comparison change with Quarkus community.

File: extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java [hunk 3]
```diff
@@ -68,37 +73,54 @@
...
 
         LOGGER.info(format("MongoDB connection string: [%s]", connectionString));
-        return connectionString;
+
+        properties.put(MONGO_DB_CONNECTION_STRING, connectionString);
+    }
+
+    private void setUsername(ServiceBinding binding, Map<String, String> properties) {
+        String username = getDbProperty(binding, DB_USER); // encodeIfContainsSpecialCharacters(getDbProperty(binding, DB_USER));
+        LOGGER.info(format("MongoDB username=%s", username));
```
Comment (geoand) [inline extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java:117]: Are you sure these need to be INFO?

**Review by geoand — COMMENTED** [review]
Any idea why the `quarkus-integration-test-mongodb-rest-data-panache` which uses service binding fails?
Are the properties I had added there not correct? If so, please update them

File: extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java (hunk 3 above)
Comment (sgahlot) [reply extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java:117]: Not really. They can be at the `debug` level too. I wasn't sure about the convention, hence, I added them at the `info` level. If that's not desirable then I can change it to `debug`.

Comment (geoand) [reply extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java:117]: Yeah, let's make them debug please

Comment (sgahlot) [reply extensions/mongodb-client/runtime/src/main/java/io/quarkus/mongodb/runtime/MongoServiceBindingConverter.java:117]: Let me also look at the integration tests as well. Could be because the connection string property doesn't contain username or password anymore.

**Comment by geoand** [comment]
When done, please squash the commits

**Review by geoand — APPROVED** [review]
